src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -1,31 +1,146 @@
+        if (goalsList.size() != 0) {
+            return goalsList.stream()
+                    .map(this::convertToDto)
+                    .collect(Collectors.toList());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/mygoals/{id}")
+    public GoalDto getGoalDtoByGoalId(@PathParam("id") long goalId) {
+        Optional<Goal> goal = goalStore.getGoalById(goalId);
+        if (goal.isPresent()) {
+            Goal g = goal.get();
+            return convertToDto(g);
+        } else {
+            return new GoalDto();
+        }
+    }
+
     @POST
     @RolesAllowed({"ADMIN", "USER"})
-    public void startPage() {
-        /// place for methods on page reload
-        /// currently - none
+    @Path("/newgoal")
+    public void createNewGoal(GoalFormDto goalDto) {
+        User user = userStore.getCurrentUser();
+        Goal goal = new Goal();
+        if (!goalDto.getGoalMessage().isEmpty() && !goalDto.getDeadline().isEmpty()) {
+            goal.setGoalMessage(goalDto.getGoalMessage());
+            goal.setTags(parseStringToTags(goalDto.getGoalMessage()));
+
+            DateTimeFormatter formatterD = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+            LocalDate localDate = LocalDate.parse(goalDto.getDeadline(), formatterD);
+            goal.setDeadlineDate(localDate);
+
+            goal.setUser(user);
+            goal.setRegisteredDate(LocalDateTime.now());
+            goalStore.addGoal(goal);
+        }
+    }
+
+    GoalDto convertToDto(Goal goal) {
+        GoalDto dto = new GoalDto();
+        dto.setUsername(goal.getUser().getUsername());
+        dto.setGoalMessage(goal.getGoalMessage());
+        dto.setDeadlineDate(convertDate(goal.getDeadlineDate()));
+        dto.setRegisteredDate(convertDateTime(goal.getRegisteredDate()));
+        dto.setDaysLeft(countDaysLeft(goal.getDeadlineDate()));
+        dto.setId(goal.getId());
+        return dto;
+    }
+
+    long countDaysLeft(LocalDate deadlineDate) {
+        LocalDate localDate = LocalDate.now();
+        return DAYS.between(localDate, deadlineDate);
+    }
+
+    String convertDate(LocalDate date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        return date.format(formatter);
+    }
+
+    String convertDateTime(LocalDateTime date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy kk:mm");
+        return date.format(formatter);
+    }
+
+    Set<Tag> parseStringToTags(String value) {
+        List<String> tagList = generateTagsList(value);
+        Set<Tag> tagSet = new HashSet<>();
+
+        for (String item : tagList) {
+            Tag tag;
+            tag = goalStore.addTag(item);
+            if (tag != null) {
+                tagSet.add(tag);
+            }
+        }
+        return tagSet;
+    }
+
+    private String[] patternList = new String[]{
```

Comment:
This list and the following method look like yet another piece of functionality that should be in its own class.

Reply:
Done